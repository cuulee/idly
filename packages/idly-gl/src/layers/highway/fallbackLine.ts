import { IDLY_NS } from '../../constants';
import { HIGHWAY } from '../priorities';

export default [
  {
    selectable: true,
    priority: HIGHWAY.MINUS_2,
    layer: {
      id: 'highway-fallback-line',
      source: undefined,
      type: 'line',
      layout: {
        'line-join': 'round',
        'line-cap': 'round',
      },
      paint: {
        'line-color': '#eee',
        'line-width': 2,
      },
      filter: ['all', ['==', `${IDLY_NS}geometry`, 'line']],
    },
  },
];
