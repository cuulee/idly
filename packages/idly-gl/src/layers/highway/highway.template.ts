const lineWidth = [
  'interpolate',
  ['exponential', 2],
  ['zoom'],
  14,
  4,
  18,
  10,
  20,
  12,
];
export const highwayTemplate: {
  selectable: true;
  layer: {
    id: undefined;
    type: 'line';
    source: undefined;
    layout: {
      'line-join': 'round';
      'line-cap': 'round';
    };
    paint: {
      'line-color': '#C0C0C0';
      'line-opacity': 1;
      'line-width': [
        'interpolate',
        ['exponential', 2],
        ['zoom'],
        14,
        4,
        18,
        10,
        20,
        12
      ];
    };
    filter: undefined;
  };
} = {
  selectable: true,
  layer: {
    id: undefined,
    type: 'line',
    source: undefined,
    layout: {
      'line-join': 'round',
      'line-cap': 'round',
    },
    paint: {
      'line-color': '#C0C0C0',
      'line-opacity': 1,
      'line-width': makeLineWidth(),
    },
    filter: undefined,
  },
};

export const highwayCaseTemplate: {
  selectable: false;
  layer: {
    id: undefined;
    type: 'line';
    source: undefined;
    layout: {
      'line-join': 'round';
      'line-cap': 'round';
    };
    paint: {
      'line-color': '#000';
      'line-opacity': 1;
      'line-width': [
        'interpolate',
        ['exponential', 2],
        ['zoom'],
        14,
        6,
        18,
        12,
        20,
        16
      ];
    };
    filter: undefined;
  };
} = {
  selectable: false,
  layer: {
    id: undefined,
    type: 'line',
    source: undefined,
    layout: {
      'line-join': 'round',
      'line-cap': 'round',
    },
    paint: {
      'line-color': '#000',
      'line-opacity': 1,
      'line-width': makeLineWidth(1.3),
    },
    filter: undefined,
  },
};

export function makeLineWidth(factor = 1) {
  return [
    ...lineWidth.slice(0, 3),
    ...(lineWidth.slice(3) as number[]).map(
      (r, i) => (i % 2 === 0 ? r : r * factor)
    ),
  ];
}
