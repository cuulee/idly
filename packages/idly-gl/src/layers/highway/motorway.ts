import { PLUGIN_NAME } from '../../constants';

export default [
  {
    selectable: true,
    priority: 2,
    layer: {
      id: 'highwayMotorway',
      type: 'line',
      source: undefined,
      layout: {
        'line-join': 'round',
        'line-cap': 'round'
      },
      paint: {
        'line-color': '#CF2081',
        'line-opacity': 1,
        'line-width': 10
      },
      filter: [
        'all',
        [
          'in',
          `${PLUGIN_NAME}--tagsClassType`,
          'tag-highway-motorway',
          /**
           * @TOFIX iD uses a mix of x_link and x-link.
           */
          'tag-highway-motorway_link'
        ]
      ]
    }
  }
];
