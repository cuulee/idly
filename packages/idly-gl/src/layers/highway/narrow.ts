import { PLUGIN_NAME } from '../../constants';

export default [
  {
    selectable: true,
    priority: 2,
    layer: {
      id: 'highwayTrack',
      type: 'line',
      source: undefined,
      layout: {
        'line-join': 'round',
        'line-cap': 'round'
      },
      paint: {
        'line-color': '#c5b59f',
        'line-opacity': 1,
        'line-width': 8
      },
      filter: [
        'all',
        ['in', `${PLUGIN_NAME}--tagsClassType`, 'tag-highway-track']
      ]
    }
  },
  {
    selectable: true,
    priority: 2,
    layer: {
      id: 'highwayNarrow',
      type: 'line',
      source: undefined,
      layout: {
        'line-join': 'round',
        'line-cap': 'round'
      },
      paint: {
        'line-color': '#c5b59f',
        'line-opacity': 0.7,
        'line-width': 5
      },
      filter: [
        'all',
        [
          'in',
          `${PLUGIN_NAME}--tagsClassType`,
          'tag-highway-path',
          'tag-highway-footway',
          'tag-highway-bridleway',
          'tag-highway-cycleway'
        ]
      ]
    }
  },
  {
    selectable: false,
    priority: 2.5,
    layer: {
      id: 'highwayNarrowCasing',
      type: 'line',
      source: undefined,
      layout: {
        'line-join': 'round',
        'line-cap': 'round'
      },
      paint: {
        'line-color': '#FFF',
        'line-opacity': 1,
        'line-width': {
          base: 1,
          stops: [[8, 3], [12, 5]]
        },
        'line-dasharray': [0.4, 2]
      },
      filter: [
        'all',
        [
          'in',
          `${PLUGIN_NAME}--tagsClassType`,
          'tag-highway-path',
          'tag-highway-footway',
          'tag-highway-bridleway',
          'tag-highway-cycleway'
        ]
      ]
    }
  }
];
