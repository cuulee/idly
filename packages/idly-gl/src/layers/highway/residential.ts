import { IDLY_NS } from '../../constants';
import { HIGHWAY } from '../priorities';
import {
  highwayCaseTemplate,
  highwayTemplate,
  makeLineWidth,
} from './highway.template';
const filter = [
  'all',
  [
    'in',
    `${IDLY_NS}tag-highway`,
    'tag-highway-residential',
    'tag-highway-residential_link',
    'tag-highway-service',
  ],
  ['!has', `${IDLY_NS}tag-service`],
];

export default [
  {
    selectable: true,
    priority: HIGHWAY.ZERO,
    layer: {
      id: 'highway-residential',
      type: 'line',
      source: undefined,
      layout: highwayTemplate.layer.layout,
      paint: {
        ...highwayTemplate.layer.paint,
        'line-color': '#FFF',
        'line-width': makeLineWidth(0.8),
      },
      filter,
    },
  },

  {
    selectable: false,
    priority: HIGHWAY.MINUS_1,
    layer: {
      id: 'highway-residential-casing',
      type: 'line',
      source: undefined,
      layout: highwayCaseTemplate.layer.layout,
      paint: {
        ...highwayCaseTemplate.layer.paint,
        'line-width': makeLineWidth(1),
      },
      filter,
    },
  },
];
