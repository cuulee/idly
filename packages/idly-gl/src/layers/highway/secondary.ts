import { IDLY_NS } from '../../constants';
import { HIGHWAY } from '../priorities';
import {
  highwayCaseTemplate,
  highwayTemplate,
  makeLineWidth,
} from './highway.template';
const filter = [
  'all',
  [
    'in',
    `${IDLY_NS}tag-highway`,
    'tag-highway-secondary',
    'tag-highway-secondary_link',
  ],
];

export default [
  {
    selectable: true,
    priority: HIGHWAY.ZERO,
    layer: {
      id: 'highway-secondary',
      type: 'line',
      source: undefined,
      layout: highwayTemplate.layer.layout,
      paint: {
        ...highwayTemplate.layer.paint,
        'line-color': '#F3F312',
        'line-width': makeLineWidth(0.8),
      },
      filter,
    },
  },
  {
    selectable: false,
    priority: HIGHWAY.MINUS_1,
    layer: {
      id: 'highway-secondary-casing',
      type: 'line',
      source: undefined,
      layout: highwayCaseTemplate.layer.layout,
      paint: {
        ...highwayCaseTemplate.layer.paint,
        'line-width': makeLineWidth(1),
      },
      filter,
    },
  },
];
