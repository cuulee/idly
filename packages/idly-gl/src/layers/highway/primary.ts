import { IDLY_NS } from '../../constants';
import { HIGHWAY } from '../priorities';
import { highwayCaseTemplate, highwayTemplate } from './highway.template';
const filter = [
  'all',
  [
    'in',
    `${IDLY_NS}tag-highway`,
    'tag-highway-primary',
    'tag-highway-primary_link',
  ],
];

export default [
  {
    selectable: true,
    priority: HIGHWAY.ZERO,
    layer: {
      id: 'highway-primary',
      type: 'line',
      source: undefined,
      layout: highwayTemplate.layer.layout,
      paint: {
        ...highwayTemplate.layer.paint,
        'line-color': '#F99806',
      },
      filter,
    },
  },
  {
    selectable: false,
    priority: HIGHWAY.MINUS_1,
    layer: {
      id: 'highway-primary-casing',
      type: 'line',
      source: undefined,
      layout: highwayCaseTemplate.layer.layout,
      paint: {
        ...highwayCaseTemplate.layer.paint,
      },
      filter,
    },
  },
];
