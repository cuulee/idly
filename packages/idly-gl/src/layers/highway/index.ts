import fallbackLine from './fallbackLine';
import motorway from './motorway';
import narrow from './narrow';
import primary from './primary';
import residential from './residential';
import secondary from './secondary';
import service from './service';
import tertiary from './tertiary';
import track from './track';
import trunk from './trunk';
import unclassified from './unclassified';

import { Layer } from '../types';

const layers: Layer[] = [
  ...fallbackLine,
  ...motorway,
  ...track,
  ...narrow,
  ...primary,
  ...residential,
  ...secondary,
  ...service,
  ...tertiary,
  ...trunk,
  ...unclassified,
];

export default layers;
