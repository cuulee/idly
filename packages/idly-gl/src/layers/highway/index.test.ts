import highway from './';
describe('highway layers snapshot', () => {
  const layerIds = highway.map(r => r.layer.id);

  it('check the structure', () => {
    expect(highway).toHaveLength(19);
    expect(highway.map(r => r.layer.id)).toMatchSnapshot();
  });

  for (const layer of layerIds) {
    it('checks layer', () => {
      expect(highway.find(r => r.layer.id === layer)).toMatchSnapshot();
    });
  }
});
