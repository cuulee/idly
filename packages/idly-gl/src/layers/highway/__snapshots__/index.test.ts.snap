// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highway layers snapshot check the structure 1`] = `
Array [
  "highway-fallback-line",
  "highway-motorway",
  "highway-motorway-casing",
  "highway-track",
  "highway-track-casing",
  "highway-narrow",
  "highway-narrow-casing",
  "highway-blue-narrow",
  "highway-blue-narrow-casing",
  "highway-green-narrow",
  "highway-green-narrow-casing",
  "highway-primary",
  "highway-primary-casing",
  "highway-residential",
  "highway-residential-casing",
  "highway-secondary",
  "highway-secondary-casing",
  "highway-service",
  "highway-service-casing",
  "highway-tertiary",
  "highway-tertiary-casing",
  "highway-trunk",
  "highway-trunk-casing",
  "highway-unclassified",
  "highway-unclassified-casing",
]
`;

exports[`highway layers snapshot checks layer 1`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "@idly-geometry",
        "line",
      ],
    ],
    "id": "highway-fallback-line",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#eee",
      "line-width": 2,
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2.5,
  "selectable": true,
}
`;

exports[`highway layers snapshot checks layer 2`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-motorway",
        "tag-highway-motorway_link",
      ],
    ],
    "id": "highway-motorway",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#CF2081",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        4,
        18,
        10,
        20,
        12,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 3,
  "selectable": true,
}
`;

exports[`highway layers snapshot checks layer 3`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-motorway",
        "tag-highway-motorway_link",
      ],
    ],
    "id": "highway-motorway-casing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#000",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        5.2,
        18,
        13,
        20,
        15.600000000000001,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2.75,
  "selectable": false,
}
`;

exports[`highway layers snapshot checks layer 4`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-track",
      ],
    ],
    "id": "highway-track",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#c5b59f",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        1.6,
        18,
        4,
        20,
        4.800000000000001,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 3,
  "selectable": true,
}
`;

exports[`highway layers snapshot checks layer 5`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-track",
      ],
    ],
    "id": "highway-track-casing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#000",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        2.2,
        18,
        5.5,
        20,
        6.6000000000000005,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2.75,
  "selectable": false,
}
`;

exports[`highway layers snapshot checks layer 6`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "@idly-geometry",
        "line",
      ],
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-path",
        "tag-highway-footway",
        "tag-highway-bridleway",
      ],
    ],
    "id": "highway-narrow",
    "layout": Object {
      "line-cap": "square",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#c5b59f",
      "line-dasharray": Array [
        3,
        3,
      ],
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        1.4,
        18,
        3.5,
        20,
        4.199999999999999,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 3,
  "selectable": true,
}
`;

exports[`highway layers snapshot checks layer 7`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "@idly-geometry",
        "line",
      ],
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-path",
        "tag-highway-footway",
        "tag-highway-bridleway",
      ],
    ],
    "id": "highway-narrow-casing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#000",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        1.8,
        18,
        4.5,
        20,
        5.4,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2.75,
  "selectable": false,
}
`;

exports[`highway layers snapshot checks layer 8`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "@idly-geometry",
        "line",
      ],
      Array [
        "all",
        Array [
          "==",
          "@idly-tag-highway",
          "tag-highway-cycleway",
        ],
      ],
    ],
    "id": "highway-blue-narrow",
    "layout": Object {
      "line-cap": "square",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#58a9ed",
      "line-dasharray": Array [
        3,
        3,
      ],
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        1.4,
        18,
        3.5,
        20,
        4.199999999999999,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 3,
  "selectable": true,
}
`;

exports[`highway layers snapshot checks layer 9`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "@idly-geometry",
        "line",
      ],
      Array [
        "all",
        Array [
          "==",
          "@idly-tag-highway",
          "tag-highway-cycleway",
        ],
      ],
    ],
    "id": "highway-blue-narrow-casing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#000",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        1.8,
        18,
        4.5,
        20,
        5.4,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2.75,
  "selectable": false,
}
`;

exports[`highway layers snapshot checks layer 10`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "@idly-geometry",
        "line",
      ],
      Array [
        "any",
        Array [
          "in",
          "@idly-tag-highway",
          "tag-highway-corridor",
          "tag-highway-pedestrian",
        ],
        Array [
          "has",
          "@idly-tag-pedestrian",
        ],
        Array [
          "has",
          "@idly-tag-corridor",
        ],
      ],
    ],
    "id": "highway-green-narrow",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#8cd05f",
      "line-dasharray": Array [
        3,
        3,
      ],
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        1.4,
        18,
        3.5,
        20,
        4.199999999999999,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 3,
  "selectable": true,
}
`;

exports[`highway layers snapshot checks layer 11`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "==",
        "@idly-geometry",
        "line",
      ],
      Array [
        "any",
        Array [
          "in",
          "@idly-tag-highway",
          "tag-highway-corridor",
          "tag-highway-pedestrian",
        ],
        Array [
          "has",
          "@idly-tag-pedestrian",
        ],
        Array [
          "has",
          "@idly-tag-corridor",
        ],
      ],
    ],
    "id": "highway-green-narrow-casing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#000",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        1.8,
        18,
        4.5,
        20,
        5.4,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2.75,
  "selectable": false,
}
`;

exports[`highway layers snapshot checks layer 12`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-primary",
        "tag-highway-primary_link",
      ],
    ],
    "id": "highway-primary",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#F99806",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        4,
        18,
        10,
        20,
        12,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 3,
  "selectable": true,
}
`;

exports[`highway layers snapshot checks layer 13`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-primary",
        "tag-highway-primary_link",
      ],
    ],
    "id": "highway-primary-casing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#000",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        5.2,
        18,
        13,
        20,
        15.600000000000001,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2.75,
  "selectable": false,
}
`;

exports[`highway layers snapshot checks layer 14`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-residential",
        "tag-highway-residential_link",
        "tag-highway-service",
      ],
      Array [
        "!has",
        "@idly-tag-service",
      ],
    ],
    "id": "highway-residential",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#FFF",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        3.2,
        18,
        8,
        20,
        9.600000000000001,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 3,
  "selectable": true,
}
`;

exports[`highway layers snapshot checks layer 15`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-residential",
        "tag-highway-residential_link",
        "tag-highway-service",
      ],
      Array [
        "!has",
        "@idly-tag-service",
      ],
    ],
    "id": "highway-residential-casing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#000",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        4,
        18,
        10,
        20,
        12,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2.75,
  "selectable": false,
}
`;

exports[`highway layers snapshot checks layer 16`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-secondary",
        "tag-highway-secondary_link",
      ],
    ],
    "id": "highway-secondary",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#F3F312",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        3.2,
        18,
        8,
        20,
        9.600000000000001,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 3,
  "selectable": true,
}
`;

exports[`highway layers snapshot checks layer 17`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-secondary",
        "tag-highway-secondary_link",
      ],
    ],
    "id": "highway-secondary-casing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#000",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        4,
        18,
        10,
        20,
        12,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2.75,
  "selectable": false,
}
`;

exports[`highway layers snapshot checks layer 18`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-service",
      ],
      Array [
        "has",
        "@idly-tag-service",
      ],
    ],
    "id": "highway-service",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#dcd9b9",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        2.8,
        18,
        7,
        20,
        8.399999999999999,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 3.25,
  "selectable": true,
}
`;

exports[`highway layers snapshot checks layer 19`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-service",
      ],
      Array [
        "has",
        "@idly-tag-service",
      ],
    ],
    "id": "highway-service-casing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#000",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        3.6,
        18,
        9,
        20,
        10.8,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 3,
  "selectable": false,
}
`;

exports[`highway layers snapshot checks layer 20`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-tertiary",
        "tag-highway-tertiary_link",
      ],
    ],
    "id": "highway-tertiary",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#FFF9B3",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        3.2,
        18,
        8,
        20,
        9.600000000000001,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 3,
  "selectable": true,
}
`;

exports[`highway layers snapshot checks layer 21`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-tertiary",
        "tag-highway-tertiary_link",
      ],
    ],
    "id": "highway-tertiary-casing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#000",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        4,
        18,
        10,
        20,
        12,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2.75,
  "selectable": false,
}
`;

exports[`highway layers snapshot checks layer 22`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-trunk",
        "tag-highway-trunk_link",
      ],
    ],
    "id": "highway-trunk",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#D54F1D",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        4,
        18,
        10,
        20,
        12,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 3,
  "selectable": true,
}
`;

exports[`highway layers snapshot checks layer 23`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-trunk",
        "tag-highway-trunk_link",
      ],
    ],
    "id": "highway-trunk-casing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#000",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        5.2,
        18,
        13,
        20,
        15.600000000000001,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2.75,
  "selectable": false,
}
`;

exports[`highway layers snapshot checks layer 24`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-unclassified",
      ],
    ],
    "id": "highway-unclassified",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#dcd9b9",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        1.6,
        18,
        4,
        20,
        4.800000000000001,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 3,
  "selectable": true,
}
`;

exports[`highway layers snapshot checks layer 25`] = `
Object {
  "layer": Object {
    "filter": Array [
      "all",
      Array [
        "in",
        "@idly-tag-highway",
        "tag-highway-unclassified",
      ],
    ],
    "id": "highway-unclassified-casing",
    "layout": Object {
      "line-cap": "round",
      "line-join": "round",
    },
    "paint": Object {
      "line-color": "#000",
      "line-opacity": 1,
      "line-width": Array [
        "interpolate",
        Array [
          "exponential",
          2,
        ],
        Array [
          "zoom",
        ],
        14,
        2.4,
        18,
        6,
        20,
        7.199999999999999,
      ],
    },
    "source": undefined,
    "type": "line",
  },
  "priority": 2.75,
  "selectable": false,
}
`;
